domenetjenester/iay/src/main/java/no/nav/foreldrepenger/abakus/domene/iay/InntektArbeidYtelseRepository.java

```diff
@@ -638,18 +652,15 @@ private Optional<InntektArbeidYtelseGrunnlag> getAktivtInntektArbeidGrunnlag(Kob
         throw new IllegalStateException("Finner flere aktive grunnlag på koblingReferanse=" + koblingReferanse);
     }
 
-    private Optional<Kobling> validerKoblingErAktiv(KoblingReferanse koblingReferanse) {
-        var kobling = koblingRepository.hentForKoblingReferanse(koblingReferanse);
-        kobling.ifPresent(InntektArbeidYtelseRepository::validerIkkeAvsluttet);
-        return kobling;
-    }
-
-    private static void validerIkkeAvsluttet(Kobling kobling) {
-        if (!kobling.erAktiv()) {
-            throw new TekniskException("FT-49000", String.format(
-                "Ikke tillatt å gjøre endringer på en avsluttet kobling. Gjelder kobling med referanse %s",
-                kobling.getKoblingReferanse()));
-        }
+    public void slettAlleInaktiveGrunnlagFor(KoblingReferanse koblingReferanse) {
+        final var query = entityManager.createQuery("""
+            DELETE FROM InntektArbeidGrunnlag gr
+            WHERE gr.aktiv = false
+            AND gr.koblingId = (SELECT k.id FROM Kobling k where k.koblingReferanse = :ref)
```

Comment:
Funker det med = eller trengs det en "IN" ?

Reply:
`SELECT k.id FROM Kobling k where k.koblingReferanse = :ref` bør være unik alltid.
Så man vil slette alle inaktive grunnlag av samme kobling(behandling). Etter sletting bør kobling - grunnlag være en 1 - 1 relasjon.